plugin/group/scaled.go

```diff
@@ -38,34 +38,42 @@ func (s *scaledGroup) changeSettings(settings groupSettings) {
 	s.settings = settings
 }
 
-func (s *scaledGroup) CreateOne(logicalID *instance.LogicalID) {
+// latestSettings gives a point-in-time view of the settings for this group.  This allows other functions to
+// safely use settings and make calls to plugins without holding the lock.
+func (s *scaledGroup) latestSettings() groupSettings {
```

Comment:
What you're describing is a different policy, and would have the result of blocking writes (updates) while the read lock is held.  As it stands, the patch decouples locking of the settings from the code acting on the settings.  My intention was to not allow long-running RPCs to block other operations.